hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/DDISimulatedDevice.java

```diff
@@ -233,4 +250,9 @@ private DdiActionFeedback calculateFeedback(final AbstractSimulatedDevice device
         }
         return feedback;
     }
+
+    private long getActionId(final Link link) {
+        final String href = link.getHref();
+        return Long.parseLong(href.substring(href.lastIndexOf('/') + 1, href.indexOf('?')));
```

Comment:
it wont, since the length doesnt match it will end up in StringIndexOutOfBoundsException. But this scenario is very unlikely to happen with the confirmation and deployment base links.